pkg/destroy/aws/aws.go

```diff
@@ -185,19 +185,23 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) ([]string, erro
 	// running on the cluster creating new resources while we are attempting to delete resources, which could leak
 	// the new resources.
 	ec2Client := ec2.New(awsSession)
+	timer := time.Now()
 	err = wait.PollImmediateUntil(
 		time.Second*10,
 		func() (done bool, err error) {
-			instancesToDelete, err := o.findEC2Instances(ctx, ec2Client, deleted)
+			instancesToDelete, instancesShuttingDown, err := o.findEC2Instances(ctx, ec2Client, deleted)
```

Comment:
Without any change in complexity in the `findEC2Instances` function, you can return (1) the instances that are not terminated and (2) the instances that are not terminated and not shutting down. Then, here you do not have to append one slice to another and can instead select which slice to use.
For example,
```
instancesNotTerminated, instancesRunning, err := o.findEC2Instances(ctx, ec2Client, deleted)
instancesToDelete := instancesRunning
if long enough since last re-terminate {
  instancesToDelete = instancesNotTerminated
}
```